src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -2394,11 +2394,16 @@ static int __icap_xclbin_download(struct icap *icap, struct axlf *xclbin)
 			goto out;
 		}
 	}
-
-
-	err = icap_download_bitstream(icap, xclbin);
-	if (err)
-		goto out;
+    /* dont need to program the bitstream for flat shell */
+
+    if(xclbin->m_header.m_mode != XCLBIN_FLAT) {
+        err = icap_download_bitstream(icap, xclbin);
+        if (err)
+            goto out;
+    } else {
+        uuid_copy(&icap->icap_bitstream_uuid, &xclbin->m_header.uuid);
+        ICAP_INFO(icap, "xclbin is generated for flat shell, dont need to program the bitstream ");
```

Comment:
Please add some comments to explain why we can skip download and any check here.